src/hotspot/share/opto/vectornode.hpp

```diff
@@ -177,47 +186,59 @@ class SaturatingVectorNode : public VectorNode {
 // Vector add byte
 class AddVBNode : public VectorNode {
  public:
-  AddVBNode(Node* in1, Node* in2, const TypeVect* vt) : VectorNode(in1,in2,vt) {}
+  AddVBNode(Node* in1, Node* in2, const TypeVect* vt) : VectorNode(in1,in2,vt) {
+    add_flag(Node::Flag_is_commutative_vector_oper);
+  }
```

Comment:
So does this still work if these vector nodes are predicated which may append another mask input in future?

Reply:
Thanks @XiaohongGong , comment addressed.